src/main/java/cloud/eppo/IConfigurationStore.java

```diff
@@ -2,13 +2,14 @@
 
 import cloud.eppo.api.Configuration;
 import java.util.concurrent.CompletableFuture;
+import org.jetbrains.annotations.NotNull;
 
 /**
  * Common interface for extensions of this SDK to support caching and other strategies for
  * persisting configuration data across sessions.
  */
 public interface IConfigurationStore {
-  Configuration getConfiguration();
+  @NotNull Configuration getConfiguration();
```

Comment:
strict contract that at least an empty configuration be returned by any implementations. Instead of having to special case "what if all the async loading fails", we simply enforce starting with an empty config, at least.